src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/step_index_pattern/append_wildcard.js

```diff
@@ -0,0 +1,53 @@
+import { uiModules } from 'ui/modules';
+
+const module = uiModules.get('apps/management');
+
+/**
+ * This directive automatically appends a wildcard to the input field
+ * after the user starts typing. It lets the user delete the wildcard
+ * if necessary. If the value of the input field is set back to an empty
+ * string, the wildcard is immediately re-appended after the user starts
+ * typing. This is intended to be a UX improvement for the index pattern
+ * creation page. See https://github.com/elastic/kibana/pull/13454
+ */
+module.directive('appendWildcard', function () {
+  return {
+    require: 'ngModel',
+    link: function ($scope, $elem, $attrs, $ctrl) {
+      $scope.appendedWildcard = false;
+
+      function setElementValue(val, event, setCursorBackOneCharacter = false) {
+        event.preventDefault();
+        $elem.val(val);
+        if (setCursorBackOneCharacter) {
+          $elem[0].setSelectionRange(1, 1);
+        }
+      }
+
+      $elem.on('keydown', (e) => {
+        const charStr = String.fromCharCode(e.keyCode);
```

Comment:
MDN warns against using [KeyboardEvent.keyCode](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) and recommends using [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key). However, Chrome 51 was the first to introduce `KeyboardEvent.key`, so it'd be nice if this "gracefully degrades" in unsupported versions, perhaps by simply not intelligently adding the `*`.

Reply:
This might be related to the following behavior, if I type `Command + R` (normally refreshes the page in Chrome on macOS) it's inserting a lower-case "r":

![command-r-hijacking](https://user-images.githubusercontent.com/627123/32173212-97e75ac6-bd54-11e7-9edd-52d8829763b4.gif)

Reply:
Nice call! Addressing the first one fixes the second one